Configurations.md

```diff
@@ -2392,6 +2392,70 @@ By default this option is set as a percentage of [`max_width`](#max_width) provi
 
 See also [`max_width`](#max_width) and [`use_small_heuristics`](#use_small_heuristics)
 
+## `single_line_let_else_max_width`
+
+Maximum line length for single line let-else statements.
+See the [let-else statement section of the Rust Style Guide](https://github.com/rust-lang/rust/blob/master/src/doc/style-guide/src/statements.md#else-blocks-let-else-statements) for more details on when a let-else statement may be written on a single line.
+A value of `0` (zero) means the divergent `else block` will always be formatted over multiple lines.
+Note this occurs when `use_small_heuristics` is set to `Off`.
+
+By default this option is set as a percentage of [`max_width`](#max_width) provided by [`use_small_heuristics`](#use_small_heuristics), but a value set directly for `single_line_let_else_max_width` will take precedence.
+
+- **Default value**: `50`
+- **Possible values**: any positive integer that is less than or equal to the value specified for [`max_width`](#max_width)
+- **Stable**: Yes
+
+#### `50` (default):
+
+```rust
+fn main() {
+    let Some(x) = opt else { return };
+
+    let Some(x) = some_very_very_very_very_long_name else {
+        return;
+    };
+
+    let Some(y) = opt else {
+        return;
+    };
```

Comment:
i assume the intent of this one is to reinforce the load-bearing nature of the semi relative to the first snippet that's able to be single lined due to width. i'm not sure it's necessary, but it obviously doesn't hurt.

however, if that's indeed the motivation then what would you think about having it immediately following the one line snippet? i feel like that association would pop a bit more if they were sequential

Reply:
Yeah, that's why it's there. I'll move it.